qiskit/transpiler/passes/optimization/commutative_cancellation.py

```diff
@@ -167,6 +171,8 @@ def run(self, dag):
                     new_op = var_z_gate(total_angle)
                 elif cancel_set_key[0] == "x_rotation":
                     new_op = RXGate(total_angle)
+                else:  # pragma: no cover
+                    raise TranspilerError("impossible case")
```

Comment:
If it's impossible, can we make it `RuntimeError` so it doesn't appear to be the user's fault were it to appear?

Reply:
done (and coverage excluded via https://github.com/Qiskit/qiskit/pull/12564)